sos/report/plugins/maas.py

```diff
@@ -8,133 +8,168 @@
-        # included here to prevent automatic journald log collection.
+        'snap.maas.pebble',
     )
 
-    option_list = [
-        PluginOpt('profile-name', default='', val_type=str,
-                  desc='Name of the remote API'),
-        PluginOpt('url', default='', val_type=str,
-                  desc='URL of the remote API'),
-        PluginOpt('credentials', default='', val_type=str,
-                  desc='Credentials, or the API key')
-    ]
-
-    is_snap = False
-
-    def _has_login_options(self):
-        return self.get_option("url") and self.get_option("credentials") \
-            and self.get_option("profile-name")
-
-    def _remote_api_login(self):
-        ret = self.exec_cmd(
-            f"maas login {self.get_option('profile-name')} "
-            f"{self.get_option('url')} {self.get_option('credentials')}"
-        )
-
-        return ret['status'] == 0
-
     def _is_snap_installed(self):
         maas_pkg = self.policy.package_manager.pkg_by_name('maas')
         if maas_pkg:
             return maas_pkg['pkg_manager'] == 'snap'
         return False
 
-    def setup(self):
-        self.is_snap = self._is_snap_installed()
-        if self.is_snap:
-            self.add_cmd_output([
-                'snap info maas',
-                'maas status'
-            ])
-
-            if self.is_service("snap.maas.pebble"):
-                # Because `snap.maas.pebble` is not in the services
-                # tuple to prevent timeouts caused by log collection,
-                # service status and logs are collected here.
-                self.add_service_status("snap.maas.pebble")
-                since = self.get_option("since") or "-1days"
-                self.add_journal(units="snap.maas.pebble", since=since)
-
-            # Don't send secrets
-            self.add_forbidden_path([
-                "/var/snap/maas/current/bind/session.key",
-                "/var/snap/maas/current/http/certs/regiond-proxy-key.pem",
-            ])
+    def _get_machines_most_recent_syslog(self, dir):
+        if not os.path.exists(dir):
+            return []
+
+        # Machine messages are collected with syslog and are stored under:
+        #   $template "{{log_dir}}/rsyslog/%HOSTNAME%/%$YEAR%-%$MONTH%-%$DAY%"
+        # Collect only the most recent "%$YEAR%-%$MONTH%-%$DAY%"
+        # for each "%HOSTNAME%".
+        recent = []
+        for host_dir in os.listdir(dir):
+            host_path = os.path.join(dir, host_dir)
+            if not os.path.isdir(host_path):
+                continue
+
+            subdirs = [
+                os.path.join(host_path, d)
+                for d in os.listdir(host_path)
+                if os.path.isdir(host_path)
+            ]
+
+            if not subdirs:
+                continue
+
+            recent.append(sorted(
+                subdirs, key=lambda f: os.stat(f).st_mtime, reverse=True)[0]
+            )
+
+        return recent
```

Comment:
I think using `glob.glob` can be shorter here but since you need to apply some logic (take most recent file from each subdir only), the code sounds good to me as well.

Reply:
Thanks for the review!

I was looking for some built-in functions that I can use together with `glob.glob` but didn't find anything.
MAAS collects _machine logs_ using rsyslog and stores them in the following way:
```
/var/snap/maas/common/log/rsyslog$ tree
.
├── 32toledo
│   └── 2021-01-21
│       └── messages
├── adm21
│   └── 2021-03-03
│       └── messages
├── amco
│   ├── 2020-05-26
│   │   └── messages
│   ├── 2020-05-28
│   │   └── messages
│   └── 2020-08-06
│       └── messages
└── witty-shrimp
    └── 2021-10-05
        └── messages
```

I thought it would be great to collect just the most recent _machine log_ (e.g. for `amco` that will be `amco/2020-08-06`) as it will speed up `sos report -o maas` in most of the cases.

If `--all_logs` provided, we want to collect all the data for all the dates

Reply:
Hm, I think for a better UX I should also check `self.get_option("since")` here and return matching directories